.github/workflows/precommit-windows.yml

```diff
@@ -0,0 +1,198 @@
+    runs-on: windows-2022
+    needs:
+      - compute-projects
+    if: ${{ needs.compute-projects.outputs.test-build == 'true' }}
+    steps:
+      - name: Setup Windows
+        uses: llvm/actions/setup-windows@main
+        with:
+          arch: amd64
+
+      - name: Fetch LLVM sources
+        uses: actions/checkout@v4
+
+      - name: Setup sccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          max-size: 2G
+          variant: sccache
+          key: precommit-windows
+
+      - name: Restore sccache from previous PR run
+        uses: ./.github/workflows/pr-sccache-restore
+
+      - name: Compute projects to test
+        id: compute-projects
+        uses: ./.github/workflows/compute-projects-to-test
+
+      - name: Configure LLVM
+        shell: bash
+        run: |
+          cmake -B build -GNinja \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DLLVM_ENABLE_PROJECTS="${{ needs.compute-projects.outputs.projects }}" \
+            -DLLVM_ENABLE_ASSERTIONS=ON \
+            -DLLVM_LIT_ARGS="-v --no-progress-bar" \
+            -DCMAKE_C_COMPILER_LAUNCHER=sccache \
+            -DCMAKE_CXX_COMPILER_LAUNCHER=sccache \
+            -S llvm
+
+      - name: Build
+        shell: bash
+        run: |
+          targets="llc"
+          if echo "${{ needs.compute-projects.outputs.check-targets }}" | grep clang; then
+            targets="$targets clang"
+          fi
+          ninja -C build $targets
+
+      - name: Save sccache for next PR run
+        if: always()
+        uses: ./.github/workflows/pr-sccache-save
+
+      - name: Package Build Directory
+        shell: bash
+        run: |
+          tar -c . | zstd -T0 -c > ../llvm-project.tar.zst
+          mv ../llvm-project.tar.zst .
+
+      - name: Upload Build
+        uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 #v4.3.0
+        with:
+          name: part1
+          path: llvm-project.tar.zst
+          retention-days: 2
+
+  build-windows-part2:
+    name: "Build All"
+    needs:
+      - build-windows-part1
+    runs-on: windows-2022
+    steps:
+      - name: Setup Windows
+        uses: llvm/actions/setup-windows@main
+        with:
+          arch: amd64
+
+      - name: Setup ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          max-size: 2G
+          variant: sccache
+          key: precommit-windows
+          # The llvm-project.zip archive contains the sccache directory from
+          # part 1, so we don't need to restore the cache.
+          restore: false
+
+      - name: Download Artifact
+        uses: actions/download-artifact@6b208ae046db98c579e8a3aa621ab581ff575935 # v4.1.1
+        with:
+          pattern: part1
+
+      - name: Unpack Artifact
+        shell: bash
+        run: |
+          tar --zstd -xf llvm-project.tar.zst
+          rm llvm-project.tar.zst
+
+      - name: Build
+        shell: bash
+        run: |
+          ls
+          ninja -C build
```

Comment:
This does not make sense: why would we build it all?

Reply:
Is there a better build target to use here to build "everything else"?  We had `test-depends` before, but that was building too much (e.g. llvm unit tests, when we only care about check-flang).  Maybe we need to add `-DLLVM_INCLUDE_TESTS=OFF` when configuring so they don't get built when not needed?

Reply:
What do you mean by “everything else”?
Why does this step exists in the first place?

Reply:
The only reason we have this step is to work around the GitHub timeouts.  If we split the build into multiple jobs, then each job has its own 6 hour timeout and it's possible to fully replicate what we have with buildkite.  Currently, this patch splits the build into 3 parts:
1. Build LLVM (and clang if necessary). `ninja llc clang`
2. Build the rest of the enabled projects and runtimes (i.e "everything else"): `ninja`
3. Run the tests. `ninja $CHECK_TARGETS`

We can split up the jobs anyway want, but I think we need to have at least two.

Reply:
Think about this more, I think I have a better idea for working around the timeouts. I'm going to try something else.